File: frontend/src/services/billingKeyService.ts
import { billingKeyApi } from "@/apis/billingKeyApi";
import type {
  BillingKeyDefaultRequest,
  BillingKeyReadyInfo,
  BillingKeyRegisterRequest,
  RegisteredBillingKey,
} from "@/types/billing";

export const billingKeyService = {
  prepareIssue(signal?: AbortSignal): Promise<BillingKeyReadyInfo> {
    return billingKeyApi.ready(signal);
  },

  listRegisteredMethods(signal?: AbortSignal): Promise<RegisteredBillingKey[]> {
    return billingKeyApi.list(signal);
  },

  registerIssuedKey(
    request: BillingKeyRegisterRequest,
    signal?: AbortSignal
  ): Promise<RegisteredBillingKey> {
    return billingKeyApi.register(request, signal);
  },

  setDefaultMethod(
    request: BillingKeyDefaultRequest,
    signal?: AbortSignal
  ): Promise<RegisteredBillingKey> {
    return billingKeyApi.setDefault(request, signal);
  },
};

File: backend/src/main/java/com/spharos/payment/service/BillingKeyService.java
package com.spharos.payment.service;

import java.util.List;

import com.spharos.payment.dto.request.BillingKeyDefaultRequest;
import com.spharos.payment.dto.request.BillingKeyRegisterRequest;
import com.spharos.payment.dto.response.BillingKeyReadyResponse;
import com.spharos.payment.dto.response.BillingKeyResponse;

public interface BillingKeyService {

    BillingKeyReadyResponse ready();

    BillingKeyResponse register(BillingKeyRegisterRequest request);

    List<BillingKeyResponse> list();

    BillingKeyResponse setDefault(BillingKeyDefaultRequest request);

    BillingKeyResponse getDefault();
}
